src/utils/src/net/tap.rs

```diff
@@ -464,6 +436,101 @@ mod tests {
         tap
     }
 
+    fn concat_slices<T: Clone>(a: &[T], b: &[T]) -> Vec<T> {
+        let mut result = Vec::with_capacity(a.len() + b.len());
+        result.extend_from_slice(a);
+        result.extend_from_slice(b);
+        result
+    }
+
+    // Builds an IPv4 packet, with an UDP datagram as a payload.
+    fn build_udp_packet(payload: &[u8]) -> Vec<u8> {
+        let payload_len: u8 = payload.len() as u8;
+
+        // A 10 bytes long header because IFF_VNET_HDR flag is set when the tap is created.
+        let v_header = [0u8; 10];
+
+        // The ethernet header consisting of:
+        // 1. Destination MAC address (6 bytes).
+        // 2. Source MAC address (6 bytes).
+        // 3. EtherType (x0800 for IPv4).
+        let ethernet_header: [u8; 14] =
+            [18, 52, 86, 120, 154, 189, 18, 52, 86, 120, 154, 188, 8, 0];
+
+        let ipv4_version = [69, 0];
+
+        // The IP total length value.
+        let total_length: [u8; 2] = [
+            0,
+            IP_HEADER_LENGTH as u8 + UDP_HEADER_LENGTH as u8 + payload_len,
+        ];
+
+        // Fixed header values:
+        // Identification, Flags, Fragment Offset, Time To Live, Protocol number (17 for UDP),
+        // Header Checksum.
+        // let ipv4_header_values = [174, 247, 64, 0, 64, 17, 40, 118];
```

Comment:
Is this a leftover?

Reply:
I'll check it out

Reply:
It's now "Static header values" instead of the confusing "Fixed"